include/tscore/ink_defs.h

```diff
@@ -126,6 +122,21 @@ int ink_sys_name_release(char *name, int namelen, char *release, int releaselen)
 int ink_number_of_processors();
 int ink_login_name_max();
 
+#ifdef __cplusplus
+// Round up a value to be a multiple of m if necessary.
+//
+template <typename ArithmeticV, typename ArithmeticM>
+constexpr ArithmeticV
+ROUNDUP(ArithmeticV value, ArithmeticM m)
+{
+  ArithmeticV remainder = value % m;
+  if (remainder) {
+    value += m - remainder;
+  }
+  return value;
+}
```

Comment:
This only provides it when it's included from a C++ file. It's a little odd to remove the C version when converting to C++. This version is unambiguously better, though, since it will refuse to compile with non-integer types.

Reply:
ink_defs.h is included by .c files, but they don't use ROUNDUP.